wordpress-mobile/WordPress-Android
# Site block title settings

Fixes #13343 

This PR implements logic for setting site title and adds tests to cover it.

To test:
- Turn on the my site flag and restart the app
- Click on My Site title
- Notice the dialog
- Change your site title
- Save your change
- The title gets updates
- Open the dialog
- Cancel it 
- The app doesn't crash

PR submission checklist:
